### PR TITLE
Added more metadata to Series tracing; Improved tracing docs, tests; Added example.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by bwplotka** [comment]
Somehow stats for series on block-level has to be fixed: 
image

Works on proxy level:

image

And on Prometheus level:

image

**Comment by bwplotka** [comment]
Works now:

File: docs/tracing.md [hunk 1]
```diff
@@ -38,9 +38,37 @@ In Kubernetes it is as easy as (on Thanos sidecar example):
...
+![view](img/tracing.png)
+
+### Forcing Sampling
+
+Every request against any Thanos component's API with header `X-Thanos-Force-Tracing` will be sampled if tracing backend was configured.
+
 ## Configuration
 
-Current tracing supported backends:
+Currently supported tracing supported backends:
```
Comment (onprem) [inline docs/tracing.md:71]:
```suggestion
Currently supported tracing backends:
```

File: docs/tracing.md [hunk 2]
```diff
@@ -38,9 +38,37 @@ In Kubernetes it is as easy as (on Thanos sidecar example):
 
 At that point, anyone can use your provider by spec.
 
+See [this issue](https://github.com/thanos-io/thanos/issues/1972) to check our progress on moving to OpenTelemetry Go client library.
+
+## Usage
+
+Once tracing is enabled and sampling per backend is configured Thanos will generate traces for all gRPC and HTTP APIs thanks to generic "middlewares". Some more interesting to observe APIs like `query` or `query_range` have more low-level spans with focused metadata showing latency for important functionalities. For example Jaeger view of HTTP query_range API call might look as follows:
```
Comment (GiedriusS) [inline docs/tracing.md:45]:
```suggestion
Once tracing is enabled and sampling per backend is configured, Thanos will generate traces for all gRPC and HTTP APIs thanks to generic "middlewares". Some more interesting to observe APIs like `query` or `query_range` have more low-level spans with focused metadata showing latency for important functionalities. For example, Jaeger view of HTTP query_range API call might look as follows:
```

**Review by GiedriusS — APPROVED** [review]
Just one nit on grammar but otherwise LGTM

**Review by onprem — DISMISSED** [review]
LGTM, thanks! Just a small non-blocking nit.

**Pull request merged**